### PR TITLE
Java Gateway improvements

### PR DESCRIPTION
**Pull request by dominikl**
This PR adds some improvements to the Java gateway:

- Adds a `move` method to the DataManagerFacility, which can be used to move objects from one parent to another (e.g. annotations from one image to another or images from one dataset to another).
- Fixes some issues with mask utilitiy methods.
- Add a methods to simply get all shapes (ie ROIs) of an image.
- Fixes an issue with the `getShapes(int z, int t)` method (it was possible that the returned list contained the same shapes multiple times).
- Adds a warning to the `getPixelSize` methods to not use the returned objects to set the pixel size, but use the `setPixelSize` methods instead.

### CODE REVIEW

**Comment by dominikl** [comment]
Hm, somehow I must have lost the integration tests for the `move` method on the way...

File: src/main/java/omero/gateway/facility/DataManagerFacility.java [hunk 1]
```diff
@@ -808,4 +810,54 @@ public FileAnnotationData call() throws Exception {
...
+     * Image to another, or Images from one Dataset to another.
+     * 
+     * @param ctx
+     *            The SecurityContext
+     * @param objects
+     *            The objects to move
+     * @param from
+     *            The object to which the objects are attached now
+     * @param to
+     *            The object to which the objects should be attached to
```
Comment (mtbc) [inline src/main/java/omero/gateway/facility/DataManagerFacility.java]: don't need the last "to", it's the same as the earlier

File: src/main/java/omero/gateway/facility/DataManagerFacility.java [hunk 2]
```diff
@@ -808,4 +810,54 @@ public FileAnnotationData call() throws Exception {
...
+     * @throws DSOutOfServiceException
+     * @throws DSAccessException
+     */
+    public void move(SecurityContext ctx, List<? extends DataObject> objects,
+            DataObject from, DataObject to)
+            throws DSOutOfServiceException, DSAccessException {
+        try {
+            if (!from.getClass().equals(to.getClass()))
```
Comment (mtbc) [inline src/main/java/omero/gateway/facility/DataManagerFacility.java]: just FYI you can just use `!=` for classes

File: src/main/java/omero/gateway/util/Links.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+     * @throws IllegalArgumentException
+     * @throws InvocationTargetException
+     */
+    public static IObject setObjects(IObject link, DataObject parent,
+            DataObject child) throws ClassNotFoundException,
+            NoSuchMethodException, SecurityException, IllegalAccessException,
+            IllegalArgumentException, InvocationTargetException {
+        Class<? extends IObject> clazz = link.getClass();
+        if (parent != null) {
+            for (Method m : clazz.getSuperclass().getMethods()) {
```
Comment (mtbc) [inline src/main/java/omero/gateway/util/Links.java]: `PropertyUtils.setProperty` can be very handy for this kind of thing.

File: src/main/java/omero/gateway/util/Mask.java [hunk 4]
```diff
@@ -168,4 +181,60 @@ private static int getFirstNonZeroInt(int[][] array) {
...
     
+    /**
+     * Breaks a one-dimensional array into 'length' chunks,
+     * forming a two-dimensional array, e.g.
+     * 
+     * int[] x = 0 1 2 3 4 5 6 7 8 9 
+     * 
+     * using length = 4 will be transformed to
+     * 
+     * int[][] y =
```
Comment (mtbc) [inline src/main/java/omero/gateway/util/Mask.java]: don't know might be worth looking into formatting options so this kind of comment renders better

**Comment by mtbc** [comment]
FYI for this kind of model class fussing another trick I've employed is while loops like https://github.com/ome/omero-blitz/blob/v5.5.0-m3/src/main/java/ome/formats/importer/ImportLibrary.java#L295

Generally looks good, my comments are entirely ignorable!

**Comment by dominikl** [comment]
Thanks @mtbc , included your suggestions. Unfortunately I have no idea how I could test them with https://github.com/openmicroscopy/openmicroscopy/pull/5970 with the split repositories now...

**Comment by dominikl** [comment]
As discussed, I'm going to close this PR and open a new one with the respective changes against omero-java-gateway later.

**Pull request closed without merging**